packages/fluentui/react/src/components/Table/TableCell.tsx

```diff
@@ -1,108 +1,135 @@
+  FluentComponentStaticProps<TableCellProps> & {
+    slotClassNames: TableCellSlotClassNames;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(TableCell.displayName, context.telemetry);
+  setStart();
+  const cellRef = React.useRef<HTMLElement>();
+
+  const { children, content, truncateContent, className, design, styles, variables } = props;
+  const hasChildren = childrenExist(children);
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(TableCell.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: TableCell.displayName,
+    actionHandlers: {
+      focusCell: e => {
+        e.preventDefault();
+        cellRef.current.focus();
+      },
+      performClick: e => {
+        handleClick(e);
+      }
+    },
+    rtl: context.rtl
+  });
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false
+  const { classes, styles: resolvedStyles } = useStyles<TableCellStylesProps>(TableCell.displayName, {
+    className: TableCell.className,
+    mapPropsToStyles: () => ({
+      truncateContent
     }),
-    content: customPropTypes.every([
-      customPropTypes.disallow(['children']),
-      PropTypes.oneOfType([PropTypes.arrayOf(customPropTypes.nodeContent), customPropTypes.nodeContent])
-    ]),
-    truncateContent: PropTypes.bool
-  };
-
-  static defaultProps = {
-    accessibility: tableCellBehavior as Accessibility
-  };
-
-  cellRef = React.createRef<HTMLElement>();
-
-  actionHandlers = {
-    focusCell: e => {
-      e.preventDefault();
-      this.cellRef.current.focus();
-    },
-    performClick: e => {
-      this.handleClick(e);
-    }
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
+  const handleClick = (e: React.SyntheticEvent) => {
     if (e.currentTarget === e.target) {
-      _.invoke(this.props, 'onClick', e, this.props);
+      _.invoke(props, 'onClick', e, props);
       e.preventDefault();
     }
   };
 
-  renderComponent({ accessibility, ElementType, styles, classes, unhandledProps }: RenderResultConfig<any>): React.ReactNode {
-    const { children, content } = this.props;
-    const hasChildren = childrenExist(children);
-
-    return (
-      <Ref innerRef={this.cellRef}>
-        <ElementType
-          className={classes.root}
-          {...accessibility.attributes.root}
-          {...unhandledProps}
-          {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
-        >
-          {hasChildren
-            ? children
-            : Box.create(content, {
-                defaultProps: () => ({ as: 'div', styles: styles.content })
-              })}
-        </ElementType>
-      </Ref>
-    );
-  }
-}
+  const tableCell = (
```

Comment:
```suggestion
  const element = (
```

_nit:_ We used `element` everywhere